src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -244,7 +245,7 @@ class ReferralService(
     }
 
     update.serviceUser?.let {
-      if (it.crn != null && it.crn != referral.serviceUserCRN) {
+      if (it.crn != referral.serviceUserCRN) {
```

Comment:
fixed warning as `crn` is always non-null

Reply:
nit: it would be good to keep these types of changes in seperate commits

Reply:
separated to d580d03af85ed5df900d47678fa8bd25fccd3d49